src/starlite_saqlalchemy/init_plugin.py

```diff
@@ -200,16 +229,21 @@ def __call__(self, app_config: AppConfig) -> AppConfig:
         self.configure_compression(app_config)
         self.configure_debug(app_config)
         self.configure_exception_handlers(app_config)
-        self.configure_health_check(app_config)
         self.configure_logging(app_config)
         self.configure_openapi(app_config)
         self.configure_sentry(app_config)
         self.configure_sqlalchemy_plugin(app_config)
         self.configure_type_encoders(app_config)
         self.configure_worker(app_config)
+        # health check is explicitly configured last
+        self.configure_health_check(app_config)
 
         app_config.before_startup = lifespan.before_startup_handler
-        app_config.on_shutdown.extend([http.on_shutdown, redis.client.close])
+        app_config.on_shutdown.append(http.on_shutdown)
+        if self.config.do_cache:
```

Comment:
I think this should be `if IS_REDIS_INSTALLED`, that way if `do_cache` is `False` but `do_worker` is true, the redis pool is still closed on shutdown.